### DISCUSSION TITLE
Add cloud coverage data to forecast property in weather entity

### DISCUSSION THREAD

**Discussion by langestefan** [home-assistant/architecture#905] [Entity Models] [answered]
### Context

Many weather forecasting services such as [Met.no](https://www.home-assistant.io/integrations/met/) and [clearoutside](https://clearoutside.com/forecast/50.7/-3.52) report cloud coverage data, in percentage [%] of sky obscured. This can be useful in for example estimating how much power your heater will need to produce, or in my case as input to forecast solar power production. For these usecases simply reporting a state such as 'cloudy' give insufficient information. See also this issue: [#74970](https://github.com/home-assistant/core/issues/74970)

**Weather integrations confirmed so far to have a cloud_cover property in their API:**

- [OpenWeatherMap](https://www.home-assistant.io/integrations/openweathermap/). Via the onecall api, gives a total cloud cover percentage for each hour. 
- [Met.no](https://www.home-assistant.io/integrations/met/).  `cloud_area_fraction` (total cloud coverage) property, see [met.no data model](https://api.met.no/doc/locationforecast/datamodel)
- [Pirate-weather-ha](https://github.com/alexander0042/pirate-weather-ha), exposes all the properties of the long deprecated DarkSky API (RIP), see also: [darksky integration](https://web.archive.org/web/20230326100953/https://www.home-assistant.io/integrations/darksky#cloud_cover)
- [Open-meteo](https://www.home-assistant.io/integrations/open_meteo/), see the [API docs](https://open-meteo.com/en/docs). 
- [SHMI.se](https://www.home-assistant.io/integrations/smhi/), see discussion below

ps. The names are all slightly different, but they all refer to the same `total_cloud_coverage`. (or just `cloud_coverage` if it's clear enough)

### Proposal

I would like to add a `cloud_coverage` data field to the forecast property in the [weather entity](https://developers.home-assistant.io/docs/core/entity/weather/). 

### Consequences

This would affect all weather integrations. Only the weather integrations that report cloud coverage in percentage will populate the `cloud_coverage` data field.

**Comment by gjohansson-ST** [comment]
You need to fill up with more integrations supporting this to add it in. As far as I understand from the two above only one is in HA.
Can the other `weather` integrations support this as well?

**Reply by langestefan** [reply]
Cloud cover is a basic weather forecasting property so I expect all weather services to have it. I will try to check more API's.

Edit: I have put the integrations I have checked so far in the post.

**Reply by gjohansson-ST** [reply]
You can add SMHI to the list (while they don't have it as % it should be relatively easy to recalculate).
I think this would make sense to add.

**Reply by langestefan** [reply]
Great, thanks for letting me know.

**Comment by frenck** [comment, marked as answer]
We have discussed this architectural proposal, and are in agreement this property can be added to both the weather entity and the forecast model of the weather entity.

../Frenck

**Comment by gjohansson-ST** [comment]
PR opened: https://github.com/home-assistant/core/pull/95068

**Reply by langestefan** [reply]
Awesome, thanks!

**Reply by frenck** [reply]
That PR has been merged, so the architectural discussion can be closed. Thanks all!
